706.py

```diff
@@ -0,0 +1,49 @@
+class MyHashMap:
+
+    def __init__(self):
+        """
+        Initialize your data structure here.
+        """
+        self.space_size = 9999
+        self.space = [[] for _ in range(self.space_size)]
+
+    def put(self, key: int, value: int) -> None:
+        """
+        value will always be non-negative.
+        """
+        hashcode = key % self.space_size
+
+        for idx, [k, v] in enumerate(self.space[hashcode]):
```

Comment:
A key-value pair should be something immutable (namely, once a pair is stored in the hash map, the content of this pair should not be allowed to be modified in any way). Using a `tuple` (i.e., `(k, v)`) instead of a `list` makes more sense here.